### DISCUSSION TITLE
Add Bank Accounts for Player Characters and Departments

### DISCUSSION THREAD

**Discussion by StrawberryMoses** [unitystation/unitystation#7524] [Ideas]
# Description

Bank accounts hold money for the player character. They do not persist between rounds. Money can be used to buy cosmetic items, extra special items, or food from vending machines. Departments also have budgets that are pulled from on payday to pay their employees. Payday is every five minutes.

The budgets are split into the following departments:
- Civilian
- Engineering
- Science
- Medical
- Service
- Cargo
- Security 

On payday, each department has money added to it's budget, and then the employees are paid.
Engineering is paid a flat 3000.
Science is paid based on a bounty system. For every slime alive, the departments funds are added to based on their color (see below). They can be paid a maximum of 5000.
Sec, medical, and service bounties are paid based on the amount of crew alive and the number of monsters killed. This is also based on moodlets, so it might be worth implementing a flat rate and updating later. They can be paid a maximum of 3000.
Service is paid 100$ times per every mood / 9 for every alive crewmember.
Medical is paid 100$ * (crewmember health / max health) for each crewmember, * the crewmember's sanity divided by 100.
Security is paid 1500 * the ratio of dead crewmembers to living crewmembers, + 150 per dead monster.
Civilian budget is paid a random amount of 1 - 5 multiplied by 500.
Cargo is paid a flat rate of 500. Their budget is shared by their request console.

Employees are paid depending on their job. It is equal to a base multiplier * (the job department account balance * 0.01). If the account does not have any money in it, there is no pay.
When a payday happens, there is a text notification. This can be disabled in settings.

Job base multipliers and departments are below.

https://github.com/tgstation/tgstation/blob/2591ce17a6609002109fc8eafa48ae0479fc0b8e/code/modules/economy/account.dm

## Category
TODO

Science slime colors:
"grey" = 10

tier 1
"orange" = 100
"metal" = 100
"blue" = 100
"purple" = 100

tier 2
"dark purple" = 500
"dark blue" = 500
"green" = 500
"silver" = 500
"gold" = 500
"yellow" = 500
"red" = 500
"pink" = 500

tier 3
"cerulean" = 750
"sepia" = 750
"bluespace" = 750
"pyrite" = 750
"light pink" = 750
"oil" = 750
"adamantine" = 750

tier 4
"rainbow" = 1000

Paycheck base multipliers:
Assistant and Minimal: 1
Easy: 2
Medium: 3
Hard: 4
Command: 10

Assistant: Assistant and Civilian
Botanist: Easy and Service
Cargo Tech: Easy and Cargo
Bartender: Easy and Service
Quartermaster: Medium and Cargo
Roboticist: Easy and Sci
Janitor: Easy and Service
Curator: Easy and Civilian
Genetecist: Medium and Medical
Lawyer: Easy and Civilian
Scientist: Medium and Science
Virologist: Medium and Medical
Clown: Minimal and Service
Chemist: Medium and Medical
Doctor: Medium and Medical
Atmos Tech: Medium and Engineering
Cook: Easy and Service
Detective: Medium and Sec
Captain: Command and Sec
CMO: Command and Med
Warden: Hard and Sec
RD: Command and Sci
Engineer: Medium and Engineering
HOP: Command and Service
Mime: Minimal and Service
HOS: Command and Sec
CE: Command and Eng
Shaft Miner: Hard and Cargo
Chaplain: Easy and Civilian
Sec Officer: Hard and Sec

Example Payday:

Cargo has 1000 dollars. On payday, 500 is added to a total of 1500.
The QM is the only one in the department. His base pay is 3, which is multiplied by the total * 0.01 to give a payday of 45$. Cargo now has 1465$.

**Comment by StrawberryMoses** [comment]
Surprise! Looks like this is getting changed. Keep an eye on this pr:

https://github.com/tgstation/tgstation/pull/48032

**Comment by PerfectTangent** [comment]
given that /tg/ has reworked the system proposed here, and we have had extensive debates about what and how the currency system should work, this is being converted to a discussion.
